README.md

```diff
@@ -408,12 +408,13 @@ A `ghw.ProcessorCore` has the following fields:
   core. Note that this does *not* necessarily equate to a zero-based index of
   the core within a physical package. For example, the core IDs for an Intel Core
   i7 are 0, 1, 2, 8, 9, and 10
-* `ghw.ProcessorCore.Index` is the zero-based index of the core on the physical
-  processor package
```

Comment:
what's our process for removing fields from the API?
Do we need any special care (deprecation) ?

Reply:
In the past I've left poorly-named fields (like `Id` instead of `ID`) in the structs in case people were using the old names and then removed them after a deprecation period. However, in this case, the old `Index` field really was confusing and wrong: it stated that it was the zero-based index of the core on the physical processor package, but that was incorrect. It was *one of* the zero-based indexes of a *logical processor* associated with the physical core... and so I figure we should just remove this field entirely and force users to change their code if they were referencing `ProcessorCore.Index`. Thoughts?